includes/Framework/PaymentGateway/Payment_Gateway.php

```diff
@@ -1834,6 +1834,119 @@ protected function do_transaction( $order ) {
 		return $this->handle_single_payment_method( $response, $order );
 	}
 
+	/**
+	 * Stores gift card details as order meta.
+	 *
+	 * @since 4.2.0
+	 *
+	 * @param \Square\Models\Order $square_order
+	 * @param \WC_Order $order
+	 */
+	public function maybe_save_gift_card_order_details( $square_order, $order ) {
+		$line_items = $square_order->getLineItems();
+
+		/** @var \Square\Models\OrderLineItem */
+		foreach ( $line_items as $line_item ) {
+			if ( \Square\Models\OrderLineItemItemType::GIFT_CARD !== $line_item->getItemType() ) {
+				continue;
+			}
+
+			$gift_card_line_item_id = $line_item->getUid();
+			$gift_card_amount       = Square_Helper::number_format(
+				Money_Utility::cents_to_float(
+					$line_item->getTotalMoney()->getAmount()
+				)
+			);
+
+			$this->update_order_meta( $order, 'gift_card_line_item_id', $gift_card_line_item_id );
+			$this->update_order_meta( $order, 'gift_card_balance', $gift_card_amount );
```

Comment:
Do we need sanitization on these values or does this already happen in `update_order_meta`?

Reply:
Don't have sanitization in the `update_order_meta` function. Addressed in https://github.com/woocommerce/woocommerce-square/pull/79/commits/23283ab6e0f54c62c4a0a1e1aeb850c994479c43